subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -469,6 +470,13 @@ static int deregister_ept(const struct device *instance, void *token)
 		return -ENOENT;
 	}
 
+	/* Drain pending work items before tearing down channel.
+	 *
+	 * Note: `k_work_flush` Faults on Cortex-M33 with "illegal use of EPSR"
+	 * if `sync` is not declared static.
+	 */
+	k_work_flush(&data->mbox_work, &sync);
```

Comment:
do we need a `k_work_cancel` as well here?

Reply:
I'm not sure how buffer ownership works within the whole metal -> openamp -> ipc service chain so I preferred not to cancel anything, in case a buffer doesn't get freed.
From the Bluetooth perspective, it's also nicer if the message from the other side is also processed.